paimon-core/src/main/java/org/apache/paimon/table/AbstractFileStoreTable.java

```diff
@@ -167,6 +167,15 @@ public Identifier identifier() {
                 : identifier;
     }
 
+    @Override
+    public String uuid() {
+        if (catalogEnvironment.uuid() != null) {
+            return catalogEnvironment.uuid();
+        }
+        long earliestCreationTime = schemaManager().earliestCreationTime();
```

Comment:
will this earliestCreationTime change as we do compaction or snapshot expire?

Reply:
Never, actually we do not clean schema files, so the first schema file (schema-0) will always exist, and immutable.

So this earliestCreationTime should always be the same.